### PR TITLE
Batch Stake-Update Txns

### PR DESCRIPTION
**Pull request by 0xKrishna**

### CODE REVIEW

File: staking/handler.go [hunk 1]
```diff
@@ -269,3 +271,66 @@ func HandleMsgValidatorExit(ctx sdk.Context, msg types.MsgValidatorExit, k Keepe
...
+		}
+
+		// pull validator from store
+		validator, ok := k.GetValidatorFromValID(ctx, stakeUpdate.ID)
+		if !ok {
+			k.Logger(ctx).Error("Fetching of validator from store failed", "validatorId", stakeUpdate.ID)
+			return hmCommon.ErrNoValidator(k.Codespace()).Result()
+		}
+
+		if stakeUpdate.Nonce != validator.Nonce+1+uint64(i) {
```
Comment (0xKrishna) [inline staking/handler.go:312]: We have to make sure the stakeUpdate array is sent in asc order or nonce.

**Comment by jdkanani** [comment]
We cannot merge this since it requires a hard fork and needs to plan in the next hard fork.

**Comment by jdkanani** [comment]
@0xKrishna This requires extensive test cases for the new tx type.

**Comment by 0xKrishna** [comment]
Yes, that is understood, I am not expecting to get it merged right now, I just raised this WIP PR so that team can see the direction I am taking this task and if that is the correct way or not. This is not even complete yet, Just for reference.

**Comment by temaniarpit27** [comment]
@0xKrishna we working on this or we can close this for now and reopen it when required?

**Comment by 0xKrishna** [comment]
Okay, I will reopen once required.

**Pull request closed without merging**